packages/block-editor/src/components/editor-styles/index.js

```diff
@@ -13,26 +13,23 @@ import { useEffect } from '@wordpress/element';
  */
 import transformStyles from '../../utils/transform-styles';
 
-function EditorStyles( { styles } ) {
+export default function useEditorStyles( ref, styles ) {
 	useEffect( () => {
 		const updatedStyles = transformStyles(
 			styles,
 			'.editor-styles-wrapper'
 		);
 
+		const { ownerDocument } = ref.current;
 		const nodes = map( compact( updatedStyles ), ( updatedCSS ) => {
-			const node = document.createElement( 'style' );
+			const node = ownerDocument.createElement( 'style' );
 			node.innerHTML = updatedCSS;
-			document.body.appendChild( node );
+			ownerDocument.body.appendChild( node );
 
 			return node;
 		} );
 
 		return () =>
-			nodes.forEach( ( node ) => document.body.removeChild( node ) );
+			nodes.forEach( ( node ) => ownerDocument.body.removeChild( node ) );
 	}, [ styles ] );
```

Comment:
Prefer adding `ref` to the dependencies array just to make linter happy in the future 😅

Reply:
A ref by definition never changes, so it is pointless to add to the dependencies. It gives the false sense that it will be reevaluated when the `ref.current` changes, which is not the case.

Reply:
I agree. It's sort of a limitation to the linter rule itself. Personally, I would prefer to just stick to rule itself and just don't think about it ever. And we won't get an annoying warning after #24914 😜 .

Reply:
Ok. It's generally not done currently in master, I even remember some people reviewing code suggesting it to be removed. If we change the rules, I think we should do it in a dedicated PR adjusting all instances.

Reply:
I don't think we're _changing_ the rules, the rules are still the same, just that we are enforcing it.

`useEditorStyles` doesn't know anything about `ref` either, all it knows is that it's a variable, we only know that it's a `ref` by the naming convention. Also, this could happen, though very unlikely:

```js
const ref1 = useRef();
const ref2 = useRef();

useEditorStyles(someState ? ref1 : ref2, style);
```

In that case, we would want to re-run the effect since `ref` changes.

We don't have to put `ref` into the dependency array if the `ref` is created in the same hook, since we know now for sure that it won't change. The lint rules also won't warn us in that case.

We can totally do it in other PR though, only that I personally prefer we include this early 😅.

Reply:
Ok, I added `ref` to the dependencies